src/controllers/mainController.ts

```diff
@@ -288,35 +289,40 @@ export default class MainController implements vscode.Disposable {
         });
     }
 
+    /**
+     * Helper function to toggle SQLCMD mode
+     */
+    private async toggleSqlCmdMode(isSqlCmd: boolean): Promise<boolean> {
+        return this._outputContentProvider.toggleSqlCmd(this._vscodeWrapper.activeTextEditorUri).then(async () => {
+            await this._connectionMgr.onChooseLanguageFlavor(true, !isSqlCmd);
+            return Promise.resolve(true);
+        });
```

Comment:
Why are you not using async/await for the whole thing?

```
await this._outputContentProvider.toggleSqlCmd(this._vscodeWrapper.activeTextEditorUri);
await this._connectionMgr.onChooseLanguageFlavor(true, !isSqlCmd);
return true;
```

I also don't see the point in returning a boolean here since you never return false.